pkg/cmd/minio_app.go

```diff
@@ -0,0 +1,137 @@
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+
+	"github.com/alexellis/k3sup/pkg/config"
+	"github.com/sethvargo/go-password/password"
+	"github.com/spf13/cobra"
+)
+
+func makeInstallMinio() *cobra.Command {
+	var minio = &cobra.Command{
+		Use:          "minio",
+		Short:        "Install minio",
+		Long:         `Install minio`,
+		Example:      `  k3sup app install minio`,
+		SilenceUsage: true,
+	}
+
+	minio.Flags().Bool("update-repo", true, "Update the helm repo")
+	minio.Flags().String("access-key", "", "Provide an access key to override the pre-generated value")
+	minio.Flags().String("secret-key", "", "Provide a secret key to override the pre-generated value")
+	minio.Flags().Bool("distributed-mode", false, "Deploy Minio in Distributed Mode")
+
+	minio.RunE = func(command *cobra.Command, args []string) error {
+		kubeConfigPath := getDefaultKubeconfig()
+
+		if command.Flags().Changed("kubeconfig") {
+			kubeConfigPath, _ = command.Flags().GetString("kubeconfig")
+		}
+		updateRepo, _ := minio.Flags().GetBool("update-repo")
+
+		fmt.Printf("Using kubeconfig: %s\n", kubeConfigPath)
+
+		userPath, err := config.InitUserDir()
+		if err != nil {
+			return err
+		}
+
+		clientArch, clientOS := getClientArch()
+
+		fmt.Printf("Client: %s, %s\n", clientArch, clientOS)
+		log.Printf("User dir established as: %s\n", userPath)
+
+		os.Setenv("HELM_HOME", path.Join(userPath, ".helm"))
+
+		_, err = tryDownloadHelm(userPath, clientArch, clientOS)
+		if err != nil {
+			return err
+		}
+
+		if updateRepo {
+			err = updateHelmRepos()
+			if err != nil {
+				return err
+			}
+		}
+
+		chartPath := path.Join(os.TempDir(), "charts")
+		err = fetchChart(chartPath, "stable/minio")
+
+		if err != nil {
+			return err
+		}
+
+		overrides := map[string]string{}
+		accessKey, _ := minio.Flags().GetString("access-key")
+		secretKey, _ := minio.Flags().GetString("secret-key")
+
+		if len(accessKey) == 0 {
+			fmt.Printf("Access Key not provided, one will be generated for you")
```

Comment:
no newline at the end of these prints - the lines get joined in output 

`Access Key not provided, one will be generated for youSecret Key not provided, one will be generated for youexec:  /Users/alistairhey/.k3sup/bin/helm template minio --name minio --namespace minio --output-dir /var/folders/y2/67nr0tqd10q35gdq84_mkz2h0000gn/T/charts/minio/rendered --values /var/folders/y2/67nr0tqd10q35gdq84_mkz2h0000gn/T/charts/minio/values.yaml  --set accessKey=#### --set secretKey=######
`